src/mixins/CoWSwapOnchainOrders.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8;
+
+import "../vendored/GPv2Order.sol";
+import "../interfaces/ICoWSwapOnchainOrders.sol";
+import "../libraries/CoWSwapEip712.sol";
+
+/// @title CoW Swap Onchain Order Creator Event Emitter
+/// @author CoW Swap Developers
+contract CoWSwapOnchainOrders is ICoWSwapOnchainOrders {
```

Comment:
Why have you chosen to make it contract and not a library? 
Is it because you wanna store  the cowSwapDomainSeparator in it? 

I think we have three options:
1) Your current implementation, then we would need to use inheritance in the ethflow contract. This is okay, but maybe suboptimal.
2) Using a library and pass calculate the kecca on the fly.
3) Use the library approach and not return the order digest. It's not strictly needed. The calling contract could always calculate it on its own.

Reply:
> Is it because you wanna store the cowSwapDomainSeparator in it?

Yes, this is the only reason. Storing the precomputed domain separator is cheaper than recomputing it and we have tight margins. (Note: compared with my first draft this is now necessary because I don't hardcode the contract address anymore. I think this will be important when we want to support different chains where deterministic deployment might not work and we are going to need different addresses.)

I could definitely make a library first, where `placeOrder` also takes a `domainSeparator`, but I expect we are always going to need the order hash in the ETH flow contract, which means a field with a domain separator will be needed.
Point 3 is interesting, for this specific project we need the order hash, however for a more general contract this might not be true. I'm ok with removing it if you think it would be better handled somewhere else.

Reply:
>Point 3 is interesting, for this specific project we need the order hash, however for a more general contract this might not be true. I'm ok with removing it if you think it would be better handled somewhere else.

I actually like the encapsulating of broadcasting + orderdigest calculation. It belongs together. 

Since the inheritance is only increasing the deployment cost but does not have any-longterm effects, it might be the best solution, even though it is not so nice from a design perspective.
Currently, I am also inclined for your current solution.